### PR TITLE
Use the same CI code to build DEBs for GitHub that we use for publishing

### PR DESCRIPTION
**Pull request by BaiShaoqi**
to LaunchPad PPA
dev: https://releng.ci.gpdb.pivotal.io/teams/main/pipelines/greenplum-database-release-deb-refactor-sbai

### CODE REVIEW

File: concourse/publish_to_ppa/run.py [hunk 1]
```diff
@@ -245,13 +250,19 @@ def run_or_fail(cmd, cwd="."):
...
         release_message=os.environ["RELEASE_MESSAGE"]
     ).build()
 
     builder = DebianPackageBuilder(source_package=source_package)
     builder.build_binary()
+    cmd = ['cp', glob.glob("*.deb")[0], 'gpdb_deb_installer/']
+    Util.run_or_fail(cmd)
+
+    # if PPA_REPO is not set, do not need to publish to ppa
+    if os.environ["PPA_REPO"].strip() == 'null' or os.environ["PPA_REPO"].strip() == '':
```
[the remark this answers is not in the document]
Comment (BaiShaoqi) [reply concourse/publish_to_ppa/run.py]: Creating the source package doesn't requires gpg private key, and ppa need source package, so os.environ['PPA_REPO'] being set means we gpg private key is set, and also means to publish to ppa

File: concourse/publish_to_ppa/run_test.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def test_build_binary(self, mocked_run_or_fail):
                                               cwd="some-dir")
 
     @patch('run.Util.run_or_fail')
+    @skipIf(os.environ["PPA_REPO"].strip() == 'null' or os.environ["PPA_REPO"].strip() == '', "gpg private key only needed when publish to ppa")
```
[the remark this answers is not in the document]
Comment (BaiShaoqi) [reply concourse/publish_to_ppa/run_test.py]: ppa_repo need source package, and source package neeed gpg private key,
ppa_repo == no gpg private key provided == source package need not run == source package test should be skip

File: concourse/publish_to_ppa/run.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marksliva) [reply concourse/publish_to_ppa/run.py]: Here is one way you can safely check for this environment variable (avoiding a KeyError):
```
if (not "PPA_REPO" in os.environ) or (os.environ["PPA_REPO"].strip() == ""):
    exit()
```

[the remark this answers is not in the document]
Comment (baotingfang) [reply concourse/publish_to_ppa/run.py]: Also another way here:

```python
# if the env NAME isn't existing, return None
os.getenv('NAME')
```

**Comment by bradfordboyle** [comment]
Let's close this PR since it has gone stale. I am sure we will be re-visiting this topic in the future.

**Pull request closed without merging**